src/cloud-element-templates/properties-panel/properties/CustomProperties/NumberProperty.js

```diff
@@ -0,0 +1,54 @@
+import { FeelNumberEntry, NumberFieldEntry } from '@bpmn-io/properties-panel';
+import { useService } from 'bpmn-js-properties-panel';
+import { propertyValidator, usePropertyAccessors } from './util';
+import { PropertyDescription } from '../../../../components/PropertyDescription';
+import { PropertyTooltip } from '../../components/PropertyTooltip';
+import { useCallback } from '@bpmn-io/properties-panel/preact/hooks';
+import { isNumber } from 'min-dash';
+
+export function NumberProperty(props) {
+  const {
+    element,
+    id,
+    property
+  } = props;
+
+  const {
+    description,
+    editable,
+    label,
+    feel,
+    tooltip
+  } = property;
+
+  const Component = feel === 'optional' ? FeelNumberEntry : NumberFieldEntry;
+
+  const bpmnFactory = useService('bpmnFactory'),
+        commandStack = useService('commandStack'),
+        debounce = useService('debounceInput'),
+        translate = useService('translate');
+
+  const [ getValue, setValue ] = usePropertyAccessors(bpmnFactory, commandStack, element, property);
+
+  const validate = useCallback((value) => {
+    if (isNumber(value) && value.toString().includes('e')) {
+      return translate('Scientific Notation is disallowed');
+    }
+
+    const defaultValidator = propertyValidator(translate, property);
+    return defaultValidator(value);
+  }, [ translate, property ]);
```

Comment:
I thought initially that it doesn't work as supposed because the automatic conversion (which I love) kicked in:

https://github.com/bpmn-io/bpmn-js-element-templates/assets/28307541/b8dda174-4434-4e4d-89f3-8936e60d086c